File: pbmx_ffi/java/io/rmf/pbmx/ffi/RawToken.java
package io.rmf.pbmx.ffi;

import java.util.Arrays;
import java.util.List;
import com.sun.jna.Structure;

public class RawToken extends Structure {
    public static class ByValue extends RawToken implements Structure.ByValue {}

    public byte[] bytes = new byte[32];

    public ByValue val() {
        ByValue value = new ByValue();
        value.bytes = this.bytes;
        return value;
    }

    public RawToken ref() {
        RawToken t = new RawToken();
        t.bytes = this.bytes;
        return t;
    }

    protected List getFieldOrder() {
        return Arrays.asList(new String[] { "bytes" });
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/ffi/RawToken.kt
package io.rmf.pbmx.ffi

import com.sun.jna.Structure

open class RawToken : Structure() {

    @JvmField var bytes = ByteArray(32)

    class ByValue : RawToken(), Structure.ByValue

    fun value(): ByValue {
        val value = ByValue()
        value.bytes = this.bytes
        return value
    }

    fun ref(): RawToken {
        val t = RawToken()
        t.bytes = this.bytes
        return t
    }

    override fun getFieldOrder(): List<String> {
        return listOf("bytes")
    }
}
